packages/form/src/components/form/index.js

```diff
@@ -17,9 +17,13 @@ const Form = ( { children, name, onSubmit, ...props } ) => {
 		[ name ]
 	);
 
+	const { startSubmit } = useDispatch( STORE_NAME );
```

Comment:
I'm not really a fan of splitting `startSubmit` and `stopSubmit` between different layer of the application and would love to keep them together.  
So either both should be called on the application level (project-renderer) or `onSubmit` could return a promise and we can deal with all of it here.

Reply:
Yeah, that makes sense. Thanks for pointing it out. 
Going with the approach of returning a promise and handling the status inside the Form component.